### PR TITLE
 Remove logic which allows wires to be passed to operations as a positional argument (Issue 187)

### PR DESCRIPTION
**Pull request by riveSunder**
I've removed the ability to pass wires as a positional argument from classes based on the ```Operation``` class. To retain the ability to pass wires as a positional argument to expval calls, I've overloaded the ```__init__``` function in the ```Expectation``` class. That can't be the prettiest way to do this as the overloaded initialization function is essentially a copy of the old version, maybe there is another way to only restore the offending part that allows wires as a positional argument, while still requiring the keyword elsewhere. 

I tested the changes in examples ```Q4_quantum-GAN.ipynb``` and ```CV1_photon-redirection.ipynb```, but there is a pre-existing bug in the unit test suite where I added a unit test. 

Thanks for looking at my PR!

### CODE REVIEW

**Comment by co9olguy** [comment]
Hi @riveSunder, thanks for the PR!

I notice that there are a number of tests failing with the latest version. Is this a work-in-progress, or ready for attention on our side (perhaps to help with debugging)?

**Comment by josh146** [comment]
Hi @riveSunder, looks great!

I've just pushed a small change that merges in the current master branch, and added in the `wires` keyword argument to the tests where it was missing.

**Comment by riveSunder** [comment]
Thanks! I see now that what I couldn't figure yesterday and was keeping me from getting the test to run was the missing ```[]``` in the line:
```for t in ([TestExpval]):```
It seems to be working fine now.

File: doc/research.rst [hunk 1]
```diff
@@ -29,7 +29,7 @@ Available categories include:
 * `PennyLane Help <https://discuss.pennylane.ai/c/pennylane-help>`_: For help and advice using PennyLane
 * `PennyLane Development <https://discuss.pennylane.ai/c/pennylane-development>`_: For discussion of PennyLane development
 * `PennyLane Plugins <https://discuss.pennylane.ai/c/pennylane-plugins>`_: For discussion of the available PennyLane plugins, and plugin development
-* `Xanadu Software <https://discuss.pennylane.ai/c/xanadu-software>`_: For discussion relating to other Xanadu software projects, including `StrawberryFields <https://github.com/xanaduai/strawberryfields>`_, `QMLT <https://github.com/xanaduai/hafnian>`_, `Hafnian <https://github.com/xanaduai/hafnian>`_, and `OpenBoson <https://github.com/xanaduai/sfopenboson>`_.
+* `Xanadu Software <https://discuss.pennylane.ai/c/xanadu-software>`_: For discussion relating to other Xanadu software projects, including `StrawberryFields <https://github.com/xanaduai/strawberryfields>`_, `QMLT <https://github.com/xanaduai/QMLT>`_, `Hafnian <https://github.com/xanaduai/hafnian>`_, and `OpenBoson <https://github.com/xanaduai/sfopenboson>`_.
```
Comment (josh146) [inline doc/research.rst]: Good catch!

**Comment by quantshah** [comment]
Is this ready to be merged? @josh146 

I think once this is merged, it gets easier to implement the multi-wire expectation as you suggested.

**Comment by josh146** [comment]
From my side it is, just waiting to see if @co9olguy would like any changes.

**Pull request merged**